ui.apps/src/main/content/jcr_root/apps/etoolbox-authoring-kit/depends-on/js/dependsOnUtils.js

```diff
@@ -154,13 +159,17 @@
     /**
      * Attempts to parse the string value into a JSON object
      * @param {string} value to parse
-     * @return {Object} parsed value or an empty object in case of any exceptions
+     * @return {Object || null} parsed value or null in case of empty object or any exceptions
      */
     ns.parseSafe = function (value) {
         try {
-            return JSON.parse(value);
+            const jsonObject = JSON.parse(value);
+            if (Object.keys(jsonObject).length === 0) {
+                return null;
+            }
+            return jsonObject;
```

Comment:
`ns.parseSafe('{}') // should be  {}`
`ns.parseSafe('"hello"') // should be 'hallo'`
`ns.parseSafe('0') // should be 0`
but
`ns.parseSafe('incorrect') // undefined or null`
`ns.parseSafe('') // undefined or null`
`ns.parseSafe('""') // ''`

```suggestion
            return JSON.parse(value);
```